python/tank_vendor/shotgun_deploy/descriptor_io/git.py

```diff
@@ -0,0 +1,337 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+import os
+import copy
+import uuid
+import tempfile
+
+from ..util import subprocess_check_output, execute_git_command
+from .base import IODescriptorBase
+from ..zipfilehelper import unzip_file
+from ..errors import ShotgunDeployError
+from ...shotgun_base import ensure_folder_exists, safe_delete_file
+
+from .. import util
+
+log = util.get_shotgun_deploy_logger()
+
+class IODescriptorGit(IODescriptorBase):
```

Comment:
You mentioned that you wished the "git" type was called "git_branch". Would it be worth changing this to `IODescriptorGitTag` and making "git_branch" a valid type as well? You'd still need "git" for backward compatibility, but maybe that could be handled with a subclass that just sets a class variable to assign the type name or something like that.